packages/desktop-client/src/components/reports/CategorySpending.js

```diff
@@ -0,0 +1,161 @@
+import React, { useState, useEffect, useMemo } from 'react';
+
+import * as d from 'date-fns';
+
+import { send } from 'loot-core/src/platform/client/fetch';
+import * as monthUtils from 'loot-core/src/shared/months';
+
+import { useActions } from '../../hooks/useActions';
+import { styles } from '../../style';
+import View from '../common/View';
+
+import CategorySelector from './CategorySelector';
+import categorySpendingSpreadsheet from './graphs/category-spending-spreadsheet';
+import CategorySpendingGraph from './graphs/CategorySpendingGraph';
+import Header from './Header';
+import useReport from './useReport';
+import { fromDateRepr } from './util';
+
+function CategoryAverage() {
+  const { getCategories } = useActions();
+
+  const [categories, setCategories] = useState({});
+  const [selectedCategories, setSelectedCategories] = useState([]);
+  const [categorySelectorVisible, setCategorySelectorVisible] = useState(false);
+
+  const [allMonths, setAllMonths] = useState(null);
+
+  const [start, setStart] = useState(
+    monthUtils.subMonths(monthUtils.currentMonth(), 5),
+  );
+  const [end, setEnd] = useState(monthUtils.currentMonth());
+
+  const [numberOfMonthsAverage, setNumberOfMonthsAverage] = useState(1);
+
+  const getGraphData = useMemo(() => {
+    return categorySpendingSpreadsheet(
+      start,
+      end,
+      numberOfMonthsAverage,
+      (categories.list || []).filter(
+        category =>
+          !category.is_income &&
+          !category.hidden &&
+          selectedCategories.some(
+            selectedCategory => selectedCategory.id === category.id,
+          ),
+      ),
+    );
+  }, [start, end, numberOfMonthsAverage, categories, selectedCategories]);
+  const perCategorySpending = useReport('category_spending', getGraphData);
+
+  useEffect(() => {
+    getCategories().then(categories => {
+      setCategories(categories);
+      setSelectedCategories(categories.list);
+    });
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏same suggestion here about the `getCategories()` usage.